uint32 n_joints
uint32 n_constraints_upper
uint32 n_constraints_lower

float64[] e
float64[] de
float64[] q
float64[] dq
float64[] ddq_star

float64[] J_upper
float64[] J_lower
float64[] b_upper
float64[] rhs_fixed_term
